gossip/src/cluster_info.rs

```diff
@@ -680,6 +680,25 @@ impl ClusterInfo {
         self.my_contact_info.read().unwrap().id
     }
 
+    pub fn keypair(&self) -> RwLockReadGuard<Arc<Keypair>> {
+        self.keypair.read().unwrap()
+    }
+
+    pub fn set_keypair(&self, new_keypair: Arc<Keypair>) {
+        let id = new_keypair.pubkey();
+
+        self.gossip.write().unwrap().set_self(&id);
+        {
+            let mut instance = self.instance.write().unwrap();
+            *instance = instance.with_id(id);
```

Comment:
maybe this should update the `timestamp` as well:
https://github.com/solana-labs/solana/blob/7782a0855/gossip/src/crds_value.rs#L393

effective this instance with this keypair is starting now, and if the timestamp is not updated and the old machine had more recent starting timestamp, then this machine will abort.

Reply:
Thanks, yep quite right.  Fixed